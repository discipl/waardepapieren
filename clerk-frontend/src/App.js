import React, {Component} from 'react';
import NeedWizard from './need-wizard/NeedWizard'
import { AbundanceService } from '@discipl/abundance-service'
import EphemeralConnector from '@discipl/core-ephemeral'






class App extends Component {
    constructor(props) {
        super(props)

        this.abundance = new AbundanceService()
    }
    async componentDidMount() {
        console.log(new EphemeralConnector().getName())
        await this.abundance.getCoreAPI().registerConnector('ephemeral', new EphemeralConnector())
        let ephemeralConnector = await this.abundance.getCoreAPI().getConnector('ephemeral')
        let ephemeralEndpoint = process.env.REACT_APP_EPHEMERAL_ENDPOINT != null ? process.env.REACT_APP_EPHEMERAL_ENDPOINT : this.props.config.DEFAULT_EPHEMERAL_ENDPOINT
        let ephemeralWebsocketEndpoint = process.env.REACT_APP_EPHEMERAL_WEBSOCKET_ENDPOINT != null ? process.env.REACT_APP_EPHEMERAL_WEBSOCKET_ENDPOINT : this.props.config.DEFAULT_EPHEMERAL_WEBSOCKET_ENDPOINT
        ephemeralConnector.configure(ephemeralEndpoint, ephemeralWebsocketEndpoint, window.WebSocket)
    }

    render() {
        return (
            <NeedWizard core={this.abundance.getCoreAPI()} config={this.props.config}/>
        );
    }
}

export default App;
