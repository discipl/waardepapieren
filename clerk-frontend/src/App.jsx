import React, {Component} from 'react';
import NeedWizard from './need-wizard/NeedWizard'
import { AbundanceService } from '@discipl/abundance-service'
import EphemeralConnector from '@discipl/core-ephemeral'
import Ipv8Connector from '@discipl/core-ipv8';

class App extends Component {
    constructor(props) {
        super(props)
        this.abundance = new AbundanceService()
    }

    async componentDidMount() {
        console.log(new EphemeralConnector().getName())
        await this.abundance.getCoreAPI().registerConnector('ephemeral', new EphemeralConnector())
        let ephemeralConnector = await this.abundance.getCoreAPI().getConnector('ephemeral')
        let ephemeralEndpoint = process.env.REACT_APP_EPHEMERAL_ENDPOINT != null ? process.env.REACT_APP_EPHEMERAL_ENDPOINT : this.props.config.DEFAULT_EPHEMERAL_ENDPOINT
        let ephemeralWebsocketEndpoint = process.env.REACT_APP_EPHEMERAL_WEBSOCKET_ENDPOINT != null ? process.env.REACT_APP_EPHEMERAL_WEBSOCKET_ENDPOINT : this.props.config.DEFAULT_EPHEMERAL_WEBSOCKET_ENDPOINT
        ephemeralConnector.configure(ephemeralEndpoint, ephemeralWebsocketEndpoint, window.WebSocket)

        if (this.props.config.ENABLE_IPV8_ATTESTATION) {
            await this.abundance.getCoreAPI().registerConnector('ipv8', new Ipv8Connector())
            let ipv8Connector = await this.abundance.getCoreAPI().getConnector('ipv8')
            ipv8Connector.configure('https://localhost/api/ipv8/site')

            // Watch for verification request where the app is the requester and approve  those
            const appDid = 'did:discipl:ipv8:TGliTmFDTFBLOnj5S5VTowiIEV3QELnWusYPUBoiJ4vXtjeznhUUHdF6CbkMsP7H4PZwESO4LPHByqZHyEkjnM986nRQ2Bit76U='
            const siteDid = 'did:discipl:ipv8:TGliTmFDTFBLOpd7y5Yd9NDKxelDmsACx9I6Go2cK4v8fKBIKzp0KiEjStqcAfSHFfk7KkqbUodAH0iX3nljnd7Y+bj8b0oFUCI='
            const core = await this.abundance.getCoreAPI()

            const verificationRequestsObserve = await core.observeVerificationRequests(siteDid, { did: appDid }, {did: null, privkey: null})
            await verificationRequestsObserve.subscribe(request => {
                ipv8Connector.ipv8AttestationClient.allowVerify(request.verifier.mid, request.claim.data)
            })
        }
    }

    render() {
        return (
            <NeedWizard core={this.abundance.getCoreAPI()} config={this.props.config}/>
        );
    }
}

export default App;
