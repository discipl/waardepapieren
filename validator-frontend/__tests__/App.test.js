import React from 'react';
import renderer from 'react-test-renderer';

import App from '../App';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';

describe('App snapshot', () => {
  jest.useFakeTimers();
  beforeEach(() => {
    NavigationTestUtils.resetInternalState();
  });

  it('renders the loading screen', async () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the root without loading screen', async () => {
    const tree = renderer.create(<App skipLoadingScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
