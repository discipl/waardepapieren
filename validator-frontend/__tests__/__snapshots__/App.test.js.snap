// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot renders the loading screen 1`] = `null`;

exports[`App snapshot renders the root without loading screen 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        collapsable={false}
        pointerEvents="auto"
        removeClippedSubviews={false}
        style={
          Array [
            Object {
              "flex": 1,
              "overflow": "hidden",
            },
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "opacity": 1,
              },
            ],
          ]
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            onLayout={[Function]}
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              onMoveShouldSetResponder={[Function]}
              onMoveShouldSetResponderCapture={[Function]}
              onResponderEnd={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderReject={[Function]}
              onResponderRelease={[Function]}
              onResponderStart={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              onStartShouldSetResponderCapture={[Function]}
              style={
                Array [
                  Object {
                    "flex": 1,
                    "flexDirection": "column-reverse",
                    "overflow": "hidden",
                  },
                  Object {
                    "backgroundColor": "#000",
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <View
                  accessibilityElementsHidden={false}
                  pointerEvents="auto"
                  style={
                    Object {
                      "backgroundColor": "#E9E9EF",
                      "bottom": 0,
                      "left": 0,
                      "opacity": 1,
                      "paddingTop": 0,
                      "position": "absolute",
                      "right": 0,
                      "shadowColor": "black",
                      "shadowOffset": Object {
                        "height": 0,
                        "width": 0,
                      },
                      "shadowOpacity": 0.2,
                      "shadowRadius": 5,
                      "top": 0,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                        Object {
                          "translateY": 0,
                        },
                      ],
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "backgroundColor": "#fff",
                        "flex": 1,
                      }
                    }
                  >
                    <RCTScrollView
                      contentContainerStyle={
                        Object {
                          "paddingTop": 30,
                        }
                      }
                      style={
                        Object {
                          "backgroundColor": "#fff",
                          "flex": 1,
                        }
                      }
                    >
                      <View>
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "marginBottom": 20,
                              "marginTop": 10,
                            }
                          }
                        >
                          <Image
                            source={1}
                            style={
                              Object {
                                "height": 80,
                                "marginLeft": -10,
                                "marginTop": 3,
                                "resizeMode": "contain",
                                "width": 100,
                              }
                            }
                          />
                        </View>
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "marginHorizontal": 50,
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "color": "rgba(0,0,0,0.4)",
                                "fontSize": 14,
                                "lineHeight": 19,
                                "marginBottom": 20,
                                "textAlign": "center",
                              }
                            }
                          >
                            Development mode is enabled. 
                            This application is in beta
                          </Text>
                          <Text
                            style={
                              Object {
                                "color": "rgba(96,100,109, 1)",
                                "fontSize": 17,
                                "lineHeight": 24,
                                "textAlign": "center",
                              }
                            }
                          >
                            Welcome to the validator app
                          </Text>
                        </View>
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "alignSelf": "center",
                              "backgroundColor": "#fcfcfc",
                              "flexDirection": "row",
                              "justifyContent": "center",
                              "marginHorizontal": 50,
                              "marginTop": 20,
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "color": "rgba(0,0,0,0.55)",
                                "fontSize": 16,
                              }
                            }
                          >
                            Use strict validation
                          </Text>
                          <RCTSwitch
                            accessibilityRole="button"
                            onChange={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Array [
                                Object {
                                  "height": 31,
                                  "width": 51,
                                },
                                Object {
                                  "flex": 1,
                                },
                              ]
                            }
                            value={false}
                          />
                        </View>
                      </View>
                    </RCTScrollView>
                  </View>
                </View>
              </View>
              <View
                onLayout={[Function]}
                pointerEvents="box-none"
                style={
                  Object {
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#FFF",
                      "transform": Array [
                        Object {
                          "translateX": -750,
                        },
                      ],
                    }
                  }
                >
                  <View
                    onLayout={[Function]}
                    pointerEvents="box-none"
                    style={
                      Object {
                        "backgroundColor": "#FFF",
                        "borderBottomColor": "#A7A7AA",
                        "borderBottomWidth": 0.5,
                        "height": 64,
                        "paddingBottom": 0,
                        "paddingLeft": 0,
                        "paddingRight": 0,
                        "paddingTop": 20,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    />
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      pointerEvents="box-none"
      style={
        Object {
          "backgroundColor": "#fff",
          "borderTopColor": "rgba(0, 0, 0, .3)",
          "borderTopWidth": 0.5,
          "flexDirection": "row",
          "height": 49,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        }
      }
    >
      <View
        accessibilityLabel="Home"
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
            },
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "flexDirection": "column",
              "justifyContent": "flex-end",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "height": 29,
              },
              false,
              Object {
                "flex": 1,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "height": "100%",
                "justifyContent": "center",
                "minWidth": 25,
                "opacity": 1,
                "position": "absolute",
                "width": "100%",
              }
            }
          >
            <Text />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "height": "100%",
                "justifyContent": "center",
                "minWidth": 25,
                "opacity": 0,
                "position": "absolute",
                "width": "100%",
              }
            }
          >
            <Text />
          </View>
        </View>
        <Text
          allowFontScaling={true}
          numberOfLines={1}
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#3478f6",
              "fontSize": 11,
              "marginBottom": 1.5,
              "textAlign": "center",
            }
          }
        >
          Home
        </Text>
      </View>
      <View
        accessibilityLabel="Scan"
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
            },
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "flexDirection": "column",
              "justifyContent": "flex-end",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "height": 29,
              },
              false,
              Object {
                "flex": 1,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "height": "100%",
                "justifyContent": "center",
                "minWidth": 25,
                "opacity": 0,
                "position": "absolute",
                "width": "100%",
              }
            }
          >
            <Text />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "height": "100%",
                "justifyContent": "center",
                "minWidth": 25,
                "opacity": 1,
                "position": "absolute",
                "width": "100%",
              }
            }
          >
            <Text />
          </View>
        </View>
        <Text
          allowFontScaling={true}
          numberOfLines={1}
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#929292",
              "fontSize": 11,
              "marginBottom": 1.5,
              "textAlign": "center",
            }
          }
        >
          Scan
        </Text>
      </View>
    </View>
  </View>
</View>
`;
